enhancements/installer/aws-internal-clusters.md

```diff
@@ -0,0 +1,192 @@
+
+#### Limitations
+
+- The Kubernetes API endpoints cannot be made public after installation using some configuration in the cluster. The users will have to manually choose the public subnets from the VPC where the cluster is deployed, created public load balancer with control-plane instances as backend and also ensure the control-plane security groups allow traffic from Internet on 6443 (Kubernetes API port). Also the user will have to make sure they pick public subnets in each Availability Zone as detailed in [documentation][public-lb-to-private-instances].
+
+### Risks and Mitigations
+
+#### Public Service type Load Balancers
+
+Creating Service type Loadbalancer that are public would require users to select public subnets and tag them `kubernetes.io/cluster/<cluster-infra-id>: shared` so that cloud provider can use them to create public load balancers. The choice of public subnets is similarly strict as mentioned [limitation][#limitations].
+
+#### Public OpenShift Ingress
+
+Changing the OpenShift Ingress to be public after installation requires extra steps for users as detailed in previous [section][#public-service-type-load-balancers]
+
+## Design Details
+
+### Test Plan
+
+The containers in the CI cluster that run the installer and the e2e tests will create VPN connection to public VPN endpoint created in the network deployment in the CI AWS account. This will provide the client transparent access to otherwise internal endpoints of the newly created cluster.
+
+### Graduation Criteria
+
+*This enhancement will follow standard graduation criteria.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- Community Testing
+- Sufficient time for feedback
+- Upgrade testing from 4.3 clusters utilizing this enhancement to later releases
+- Downgrade and scale testing are not relevant to this enhancement
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+### Upgrade / Downgrade Strategy
+
+Not applicable
+
+### Version Skew Strategy
+
+Not applicable.
+
+## Implementation History
+
+## Drawbacks
+
+Customer owned networking components means the cluster cannot automatically change things about the networking during upgrade.
+
+## Alternatives
+
+Similar to the `Drawbacks` section the `Alternatives` section is used to
+highlight and record other possible approaches to delivering the value proposed
+by an enhancement.
+
+## Infrastructure Needed
+
+- Network deployment in AWS CI account which includes VPC, subnets, NAT gateways, Internet gateway that provide egress to the Internet for instance in private subnet.
+- VPN accessibility to the network deployment created above.
+
+[aws-shared-networking]: ./aws-customer-provided-subnets.md
```

Comment:
nit: no need for the `./`.

Reply:
done https://github.com/openshift/enhancements/compare/d1df0a7526e3ead64c786034f1f50e29d556a5aa..d5621ab1053e0b04209359ebcd82d867ab38ac2c